R/get_schema.R

```diff
@@ -8,11 +8,36 @@
 #' dplyr::copy_to(conn, mtcars, name = "mtcars")
 #'
 #' get_schema(conn)
-#' get_schema(get_table(conn, "mtcars"))
+#' if (table_exists(conn, id("mtcars", conn = conn))) {
+#'   get_schema(get_table(conn, id("mtcars", conn = conn)))
+#' }
 #'
 #' close_connection(conn)
 #' @export
 get_schema <- function(.x) {
+  UseMethod("get_schema")
+}
+
+#' @export
+get_schema.PqConnection <- function(.x) {
+  return(DBI::dbGetQuery(.x, "SELECT CURRENT_SCHEMA()")$current_schema)
+}
+
+#' @export
+get_schema.SQLiteConnection <- function(.x) {
+  schemata <- unique(get_tables(.x)["schema"])
+
+  if ("main" %in% schemata) {
+    return("main")
+  } else if ("temp" %in% schemata) {
+    return("temp")
+  } else {
+    return()
+  }
+}
+
+#' @export
+get_schema.default <- function(.x) {
```

Comment:
Consider coupling these through `@name` and `@rdname`

Reply:
I would argue against this, as the usage of `get_schema` is consistent across object classes and therefore clutter the help page. For reference, the help page for `print` does not list the methods for each individual registered S3 method, but only (on my machine) `factor`, `table` and `function`, as these classes (or subclasses thereof) are often used take a widely varying range of arguments.